packages/maskbook/src/plugins/Snapshot/UI/NetworkFail.tsx

```diff
@@ -0,0 +1,29 @@
+import { Component } from 'react'
+import { SnapshotCard } from './SnapshotCard'
+import { Typography, Button } from '@material-ui/core'
+
+export class NetworkFail extends Component<{ title: string; retry: () => void }> {
+    static getDerivedStateFromError(error: unknown) {
+        return { error }
+    }
+    state: { error: Error | null } = { error: null }
+    render() {
+        if (this.state.error) {
+            return (
+                <SnapshotCard title={this.props.title}>
```

Comment:
Look like this:

![image](https://user-images.githubusercontent.com/63733714/114868415-77292e00-9e28-11eb-83f5-43a419dc6be0.png)